### PR TITLE
fix: upgrade npm dependency to v8

### PR DESCRIPTION
**Pull request by AlexanderBabel**
This PR fixes the GHSA-896r-f27r-55mw and GHSA-93q8-gq69-wqmw vulnerabilities by updating `npm` to v8.

The `npm` [v8.0.0](https://github.com/npm/cli/releases/tag/v8.0.0) package drops support for Node 10 and 11. These versions are no longer supported by this project. Therefore, this is not a breaking change.

```
# npm audit report

ansi-regex  >2.1.1 <5.0.1
Severity: moderate
 Inefficient Regular Expression Complexity in chalk/ansi-regex - https://github.com/advisories/GHSA-93q8-gq69-wqmw
fix available via `npm audit fix`
node_modules/npm/node_modules/cli-table3/node_modules/ansi-regex
node_modules/npm/node_modules/string-width/node_modules/ansi-regex
  strip-ansi  4.0.0 - 5.2.0
  Depends on vulnerable versions of ansi-regex
  node_modules/npm/node_modules/string-width/node_modules/strip-ansi
    string-width  2.1.0 - 4.1.0
    Depends on vulnerable versions of strip-ansi
    node_modules/npm/node_modules/string-width

json-schema  <0.4.0
Severity: moderate
json-schema is vulnerable to Prototype Pollution - https://github.com/advisories/GHSA-896r-f27r-55mw
fix available via `npm audit fix`
node_modules/npm/node_modules/json-schema
  jsprim  0.3.0 - 1.4.1 || 2.0.0 - 2.0.1
  Depends on vulnerable versions of json-schema
  node_modules/npm/node_modules/jsprim
```

Additionally, I updated the `engine` section in the `package.json`. https://github.com/semantic-release/npm/pull/408 fails because of a miss match. I applied the recommendation from the workflow run https://github.com/semantic-release/npm/runs/4502004243.

### CODE REVIEW

**Comment by travi** [comment]
> I updated the `engine` section in the `package.json`. #408 fails because of a miss match.

it looks like [v8 of npm dropped support for node v15](https://github.com/npm/cli/commit/a13d9d53ddf3e0f52f4a39fe116653bf40cf99e5#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R240), so the need for this makes sense.

> The `npm` [v8.0.0](https://github.com/npm/cli/releases/tag/v8.0.0) package drops support for Node 10 and 11. These versions are no longer supported by this project. Therefore, this is not a breaking change.

since v15 is also dropped, it is technically a breaking change

we are currently working through [converting the project to ESM](https://github.com/semantic-release/npm/issues/436), which would also be a breaking change. along with that, we plan to raise the minimum node version to v16, which would support this npm version more naturally. 

i tried to look at the npm releases to see if these fixes are available in a v7 release, but it doesnt look promising from a brief look. could you confirm if this is resolvable without the v8 upgrade, if you havent already?

**Comment by AlexanderBabel** [comment]
There is an issue about this in the npm/cli repo: https://github.com/npm/cli/issues/3785

But it doesn't look like there is something happening regarding this issue.

File: package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
     "xo": "0.36.1"
   },
   "engines": {
-    "node": ">=14.17"
+    "node": "^17 || ^16 || ^14.17"
```
Comment (travi) [inline package.json:51]:
```suggestion
    "node": ">=16 || ^14.17"
```

lets keep the upper bound open

**Review by travi — APPROVED** [review]

**Comment by travi** [comment]
thanks a lot for the investigation and contribution, @AlexanderBabel!

**Pull request merged**